src/server/game/Battlegrounds/ArenaTeam.cpp

```diff
@@ -314,9 +315,33 @@ void ArenaTeam::SetCaptain(ObjectGuid guid)
 
 void ArenaTeam::DelMember(ObjectGuid guid, bool cleanDb)
 {
+    Player* player = ObjectAccessor::FindPlayer(guid);
+    
     // Remove member from team
     for (MemberList::iterator itr = Members.begin(); itr != Members.end(); ++itr)
     {
+        // Remove queues of members
+        if (Player* playerMember = ObjectAccessor::FindConnectedPlayer(itr->Guid))
+        {
+            if (player && player->GetGroup() && playerMember->GetGroup() && player->GetGroup()->GetGUID() == playerMember->GetGroup()->GetGUID())
```

Comment:
you could move the "player && player->GetGroup()" check outside of the for loop, maybe saving the group in a variable, so avoid the loop totally if the player is not in group.